src/tutorials/ex3.c

```diff
@@ -0,0 +1,86 @@
+
+static char help[] = "Trains binary SVM classification model using precomputed Gramian matrix.\n\
+Input parameters:\n\
+  -f       : training dataset file\n\
+  -f_kernel: file contains precomputed Gramian\n\
+  -f_test  : test dataset file\n\
+  -f_training_predictions: file to save predictions on training samples\n\
+  -f_test_predictions    : file to save predictions on test samples\n";
+
+#include <permonsvm.h>
+
+int main(int argc,char **argv)
+{
+  SVM            svm;
+
+  char           training_file[PETSC_MAX_PATH_LEN]           = "data/heart_scale.bin";
+  char           test_file[PETSC_MAX_PATH_LEN]               = "";
+  char           training_result_file[PETSC_MAX_PATH_LEN]    = "";
+  char           test_result_file[PETSC_MAX_PATH_LEN]        = "";
+  PetscBool      test_file_set = PETSC_FALSE;
+  PetscBool      training_result_file_set=PETSC_FALSE,test_result_file_set=PETSC_FALSE;
+
+  PetscViewer    viewer;
+  PetscErrorCode ierr;
+
+  ierr = PermonInitialize(&argc,&argv,(char *)0,help); if (ierr) return ierr;
+
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_training",training_file,sizeof(training_file),NULL);CHKERRQ(ierr);
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_test",test_file,sizeof(test_file),&test_file_set);CHKERRQ(ierr);
+
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_training_predictions",training_result_file,sizeof(training_result_file),&training_result_file_set);CHKERRQ(ierr);
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_test_predictions",test_result_file,sizeof(test_result_file),&test_result_file_set);CHKERRQ(ierr);
```

Comment:
Just view them into stdout. Then it will be possible to check the results from the test suite and it simplifies the code.

Reply:
I like this. Implemented.